tensorflow_datasets/image/structseg.py

```diff
@@ -0,0 +1,140 @@
+"""
+Organ-at-risk segmentation dataset from head & neck CT scans
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import re
+import tensorflow_datasets.public_api as tfds
+import tensorflow as tf
+import nibabel as nib
+import numpy
+
+_CITATION = """
+@misc{li_zhou_deng_chen, title={StructSeg2019 - Grand Challenge}, 
+url={https://structseg2019.grand-challenge.org/}, 
+journal={Automatic Structure Segmentation for Radiotherapy Planning Challenge }, 
+author={Li, Hongsheng and Zhou, Jinghao and Deng, Jincheng and Chen, Ming}}
+"""
+
+_DESCRIPTION = """
+The dataset is used for evaluating automatic algorithms on segmentation of organ-at-risk(OAR)
+of nasopharynx cancer, for radiation therapy planning. The dataset contains organ-at-risk 
+segmentation from head & neck CT scans 
+"""
+
+_BASE_URL = """https://structseg2019.grand-challenge.org/"""
+
+
+class Structseg(tfds.core.GeneratorBasedBuilder):
+    """
+    Organ-at-risk segmentation dataset from head & neck CT scans
+    """
+    # VERSION = tfds.core.Version('0.1.0')
+
+    VERSION = tfds.core.Version("1.0.0",
+                                experiments={tfds.core.Experiment.S3: False})
+    SUPPORTED_VERSIONS = [
+        tfds.core.Version("2.0.0"),
+    ]
+
+    def get_all_file_paths(self, directory):
+        """
+        Images stored in the following format:
+        HaN_OAR/1/image.nii.gz
+        HaN_OAR/1/label.nii.gz
+        HaN_OAR/2/image.nii.gz
+        HaN_OAR/2/label.nii.gz ...
+        Get the file path to all images
+        """
+
+        file_paths = []
+
+        # crawling through directory and subdirectories
+        for root, directories, files in os.walk(directory):
```

Comment:
use tf.io.gfile instead of os: https://www.tensorflow.org/api_docs/python/tf/io/gfile/listdir.
Running tests will provide you an `AssertionError` on this.

The reason is because we want to be compatible with gcs `gs://` filepaths